### PR TITLE
Ambiguity in LocalizedTaxonomyController action names

### PR DESCRIPTION
**Pull request by AndreaPiovanelli**
This fixes an ambiguity issue for the GetTaxonomy action of LocalizedTaxonomyController. There was a "new GetTaxonomy" action in AdminLocalizedTaxonomyController too that was ambiguous, failing to properly work. For this reason, the action has been removed from AdminLocalizedTaxonomyController and the admin filter is added via override of the ApplyPreRequest function.

### CODE REVIEW

**Comment by BenedekFarkas** [comment]
Can you provide repro steps to compare before/after?

**Comment by AndreaPiovanelli** [comment]
> Can you provide repro steps to compare before/after?

The issue was easily reproducible when opening the edit page of an item containing a TaxonomyField. The ajax call in TaxonomyFieldList shape (https://github.com/LaserSrl/Orchard/blob/8ed7b67f1fc865e60b1de658b085fb49efb45151/src/Orchard.Web/Modules/Orchard.Taxonomies/Views/EditorTemplates/Fields/TaxonomyFieldList.cshtml#L13) failed, showing the "Loading taxonomy failed" alert message at line 136 of the same file. The error wasn't logged by Orchard, because it's a MVC error, but you can see it in the browser console:
Loading taxonomy fail message:
image
The error displayed was the following:
The current request for action 'GetTaxonomy' on controller type 'AdminLocalizedTaxonomyController' is ambiguous between the following action methods:

> The current request for action 'GetTaxonomy' on controller type 'AdminLocalizedTaxonomyController' is ambiguous between the following action methods:
> System.Web.Mvc.ActionResult GetTaxonomy(System.String, System.String, Int32, System.String, System.String) on type Orchard.Taxonomies.Controllers.AdminLocalizedTaxonomyController
> System.Web.Mvc.ActionResult GetTaxonomy(System.String, System.String, Int32, System.String, System.String) on type Orchard.Taxonomies.Controllers.LocalizedTaxonomyController

I didn't notice this problem before PR #8780 was merged, because of the javascript syntax error blocking the proper execution of the code.

File: src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs
Comment (BenedekFarkas) [inline src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs]:
```suggestion
        protected virtual void ApplyPreRequest() { }
```

File: src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs
Comment (BenedekFarkas) [inline src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs:1]: Please apply automatic formatting and usings cleanup in both this and `AdminLocalizedTaxonomyContoller`.

**Review by BenedekFarkas — CHANGES_REQUESTED** [review]
Two small things noted, otherwise works great!

File: src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs
Comment (AndreaPiovanelli) [reply src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs:1]: Autoformatting and cleanup should be ok in last commit (https://github.com/OrchardCMS/Orchard/pull/8785/commits/2f6ada44cf1ad56d34a2f86d59d2b4f1a81ec13f)

Comment (BenedekFarkas) [reply src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs:1]: Great, thanks!

File: src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/AdminLocalizedTaxonomyController.cs
Comment (BenedekFarkas) [inline src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/AdminLocalizedTaxonomyController.cs]: Actually, I just realized I misread the class names, but with `AdminLocalizedTaxonomyController` deriving from `LocalizedTaxonomyController`, this whole thing can be made a lot easier with proper inheritance. All the suggestions need to be applied together for them to work.
```suggestion
﻿using System.Web.Mvc;
﻿﻿using System.Web.Routing;
```

File: src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/AdminLocalizedTaxonomyController.cs
Comment (BenedekFarkas) [inline src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/AdminLocalizedTaxonomyController.cs]:
```suggestion
        [OutputCache(NoStore = true, Duration = 0)]
        public override ActionResult GetTaxonomy(string contentTypeName, string taxonomyFieldName, int contentId, string culture, string selectedValues) {
            AdminFilter.Apply(_requestContext);
```

File: src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/AdminLocalizedTaxonomyController.cs
Comment (BenedekFarkas) [inline src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/AdminLocalizedTaxonomyController.cs:29]:
```suggestion
            return base.GetTaxonomy(contentTypeName, taxonomyFieldName, contentId, culture, selectedValues);
```

File: src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs
Comment (BenedekFarkas) [inline src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs]:
```suggestion
```

File: src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs
Comment (BenedekFarkas) [inline src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs]:
```suggestion
```

File: src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs
Comment (BenedekFarkas) [inline src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/LocalizedTaxonomyController.cs]:
```suggestion
        public virtual ActionResult GetTaxonomy(string contentTypeName, string taxonomyFieldName, int contentId, string culture, string selectedValues) {
```

**Review by BenedekFarkas — CHANGES_REQUESTED** [review]

**Comment by BenedekFarkas** [comment]
@AndreaPiovanelli please check the suggested changes and/or enable edits from maintainers.

**Comment by AndreaPiovanelli** [comment]
@BenedekFarkas I took some time to test your suggestions in our environment and have now applied the changes you requested to current pr branch. 
I had to apply the last one separatedly (not via the "commit suggestion" button) because git disabled that button as apparently it was an "outdated suggestion" for it.

**Comment by BenedekFarkas** [comment]
OK, thanks!

**Review by BenedekFarkas — APPROVED** [review]

**Pull request merged**
